std/range/primitives.d

```diff
@@ -1987,6 +1993,93 @@ ElementType!R moveAt(R, I)(R r, I i) if (isIntegral!I)
 }
 
 /**
+Pops the first element from $(D r) and returns it.
+It combines $(D front) and $(D popFront) in one method.
+
+Params:
+    r = Input range
+
+Returns:
+    Next element of the input range
+ */
+auto frontPop(Range)(ref Range r)
+if (isInputRange!Range)
+in
+{
+    assert(!r.empty, "r cannot be empty");
```

Comment:
The assertion message is not useful (the __r__). Either it must be removed or changed for more value added.
I suggest `assert(!r.empty, "empty range passed in " ~ __PRETTY_FUNCTION__); `. With pretty function, the type of the argument is also written.

same remark for `backPop` input contract.